File: civitas/src/civitas/Sorpresa.java
package civitas;


import java.util.ArrayList;

public abstract class Sorpresa {
    
    private String texto;
    
    
    public abstract void aplicarAJugador(int actual,  ArrayList<Jugador> todos);
    
    /**
     * Informa de la sorpresa que obtiene el jugador.
     * @param actual indice del jugador actual.
     * @param todos jugadores.
     */
    protected void informe(int actual,  ArrayList<Jugador> todos) {
        Diario.getInstance().ocurreEvento("El jugador " + 
                todos.get(actual).getNombre() + 
                " obtiene la sorpresa " + texto);
    }
   
    /**
     * Comprueba si el jugador actual es correcto.
     * @param actual indice del jugador actual.
     * @param todos jugadores.
     * @return 
     */
    public boolean jugadorCorrecto(int actual,  ArrayList<Jugador> todos) {
        return actual < todos.size() && actual >= 0;
    }
    
    Sorpresa(String texto) {
        this.texto = texto;
    }
    
    /**
     * Proporciona informacion de la sorpresa.
     * @return 
     */
    @Override
    public String toString() {
        String salida = "Sorpresa: " + texto;
        return salida;
    }
    
}

File: civitasRuby/lib/sorpresa.rb
# encoding:utf-8

require_relative 'jugador'
require_relative 'sorpresa'

module Civitas
  
  
  
  class Sorpresa
    
    def initialize(texto)
      @texto = texto
    end
    
    def aplicar_a_jugador(actual, todos)
      informe(actual, todos)
    end
    
    def jugador_correcto(actual, todos)
      return actual <= todos.size 
    end

    def informe(actual, todos)
      Diario.instance.ocurre_evento("El jugador #{todos.at(actual).nombre} 
obtiene la sorpresa #{@texto} \n")
    end
     
    def to_string
      return @texto
    end
    
  private_class_method :new
  private :informe
  end
 end
 